src/main/java/io/aiven/kafka/connect/s3/S3SinkTask.java

```diff
@@ -79,12 +80,7 @@ public void start(final Map<String, String> props) {
             AmazonS3ClientBuilder
                 .standard()
                 .withCredentials(
-                    new AWSStaticCredentialsProvider(
-                        new BasicAWSCredentials(
-                            config.getAwsAccessKeyId().value(),
-                            config.getAwsSecretKey().value()
-                        )
-                    )
+                    credentialFactory.getProvider(config)
```

Comment:
One of the main changes here. Responsibility for Credential Provider selection goes to a new class

Reply:
Yeap it is clear